src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
> The converse is probably not.

And this is not true. `ui_event` (only used for external stuff, not for TUI) is easy to grep for.